mediapipe/tasks/python/vision/image_segmenter.py

```diff
@@ -0,0 +1,237 @@
+
+  @classmethod
+  def create_from_model_path(cls, model_path: str) -> 'ImageSegmenter':
+    """Creates an `ImageSegmenter` object from a TensorFlow Lite model and the
+    default `ImageSegmenterOptions`.
+
+    Note that the created `ImageSegmenter` instance is in image mode, for
+    performing image segmentation on single image inputs.
+
+    Args:
+      model_path: Path to the model.
+
+    Returns:
+      `ImageSegmenter` object that's created from the model file and the default
+      `ImageSegmenterOptions`.
+
+    Raises:
+      ValueError: If failed to create `ImageSegmenter` object from the provided
+        file such as invalid file path.
+      RuntimeError: If other types of error occurred.
+    """
+    base_options = _BaseOptions(model_asset_path=model_path)
+    options = ImageSegmenterOptions(
+        base_options=base_options, running_mode=_RunningMode.IMAGE)
+    return cls.create_from_options(options)
+
+  @classmethod
+  def create_from_options(cls,
+                          options: ImageSegmenterOptions) -> 'ImageSegmenter':
+    """Creates the `ImageSegmenter` object from image segmenter options.
+
+    Args:
+      options: Options for the image segmenter task.
+
+    Returns:
+      `ImageSegmenter` object that's created from `options`.
+
+    Raises:
+      ValueError: If failed to create `ImageSegmenter` object from
+        `ImageSegmenterOptions` such as missing the model.
+      RuntimeError: If other types of error occurred.
+    """
+
+    def packets_callback(output_packets: Mapping[str, packet_module.Packet]):
+      if output_packets[_IMAGE_OUT_STREAM_NAME].is_empty():
+        return
+      segmentation_result = packet_getter.get_image_list(
+          output_packets[_SEGMENTATION_OUT_STREAM_NAME])
+      image = packet_getter.get_image(output_packets[_IMAGE_OUT_STREAM_NAME])
+      timestamp = output_packets[_SEGMENTATION_OUT_STREAM_NAME].timestamp
+      options.result_callback(segmentation_result, image,
+                              timestamp.value // _MICRO_SECONDS_PER_MILLISECOND)
+
+    task_info = _TaskInfo(
+        task_graph=_TASK_GRAPH_NAME,
+        input_streams=[':'.join([_IMAGE_TAG, _IMAGE_IN_STREAM_NAME])],
+        output_streams=[
+            ':'.join([_SEGMENTATION_TAG, _SEGMENTATION_OUT_STREAM_NAME]),
+            ':'.join([_IMAGE_TAG, _IMAGE_OUT_STREAM_NAME])
+        ],
+        task_options=options)
+    return cls(
+        task_info.generate_graph_config(
+            enable_flow_limiting=options.running_mode ==
+            _RunningMode.LIVE_STREAM), options.running_mode,
+        packets_callback if options.result_callback else None)
+
+  def segment(self,
+              image: image_module.Image) -> List[image_module.Image]:
+    """Performs the actual segmentation task on the provided MediaPipe Image.
+
+    Args:
+      image: MediaPipe Image.
+
+    Returns:
+      A segmentation result object that contains a list of segmentation masks
```

Comment:
nit: unlike Task Library, the return is not a `segmentation result object`. Please align the doc string with the C++ comment with explanation of CATEGORY_MASK / CONFIDENCE_MASK: https://github.com/google/mediapipe/blob/master/mediapipe/tasks/cc/vision/image_segmenter/image_segmenter.h#L114-L125

Same for the `segment_for_video` method

Reply:
Updated the docstrings for both the APIs now.